pkg/helmpath/xdg_darwin_test.go

```diff
@@ -0,0 +1,85 @@
+// Copyright The Helm Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build darwin
+
+package helmpath
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"k8s.io/client-go/util/homedir"
+)
+
+const (
+	appName  string = "helm"
+	testFile string = "test.txt"
+)
+
+var testApp = app{name: appName}
+
+func TestDataPath(t *testing.T) {
+	os.Unsetenv("XDG_DATA_HOME")
```

Comment:
Yes, in case the user has this environment variable already set. Unsetting the environment variable ensures this will work on anyone's machine. We want to test the scenario where when the environment variable is unset, we use the default home path.

The `UnsetEnv` wasn't necessary for root_test.go, so I removed that. Good catch.